### PR TITLE
fix(backend): fallback if `sinceId` is older than the oldest in cache when using FTT (#14061)

### PR DESCRIPTION
**Pull request by Sayamame-beans**
Cherry-picked from misskey-dev#14061 (8a9de08)

* fix(backend): fallback if `sinceId` is older than the oldest in cache when using FTT

* Update CHANGELOG.md

* chore: fix description of test

## Checklist
- [ ] Read the [contribution guide](https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md)
- [ ] Test working in a local environment
- [ ] (If needed) Add story of storybook
- [ ] (If needed) Update CHANGELOG.md
- [ ] (If possible) Add tests

### CODE REVIEW

File: packages/backend/src/core/FanoutTimelineEndpointService.ts [hunk 1]
```diff
@@ -67,18 +64,18 @@ export class FanoutTimelineEndpointService {
...
 		// TODO: いい感じにgetMulti内でソート済だからuniqするときにredisResultが全てソート済なのを利用して再ソートを避けたい
-		const redisResultIds = shouldFallbackToDb ? [] : Array.from(new Set(redisResult.flat(1)));
+		const redisResultIds = shouldFallbackToDb ? [] : Array.from(new Set(redisResult.flat(1))).sort(idCompare);
 
-		redisResultIds.sort(idCompare);
-		noteIds = redisResultIds.filter(id => id >= thresholdId).slice(0, ps.limit);
+		let noteIds = redisResultIds.filter(id => id >= thresholdId).slice(0, ps.limit);
 
-		shouldFallbackToDb = shouldFallbackToDb || (noteIds.length === 0);
+		const oldestNoteId = ascending ? redisResultIds[0] : redisResultIds[redisResultIds.length - 1];
```
Comment (Sayamame-beans) [inline packages/backend/src/core/FanoutTimelineEndpointService.ts:77]: `redisResultIds`が[]のとき、oldestNoteIdはundefined

File: packages/backend/src/core/FanoutTimelineEndpointService.ts [hunk 2]
```diff
@@ -67,18 +64,18 @@ export class FanoutTimelineEndpointService {
...
+		shouldFallbackToDb = shouldFallbackToDb || (noteIds.length === 0) || (ps.sinceId != null && ps.sinceId < oldestNoteId);
```
Comment (Sayamame-beans) [inline packages/backend/src/core/FanoutTimelineEndpointService.ts:78]: String < undefinedの比較はfalse

Comment (anatawa12) [reply packages/backend/src/core/FanoutTimelineEndpointService.ts:78]: upstreamの問題 && 家のサーバーで問題にならないからいいけどここの`(ps.sinceId != null && ps.sinceId < oldestNoteId)`は`ps.useDbFallback &&`取ったほうがいいかもね。

**Review by anatawa12 — APPROVED** [review]
よさそう。

**Comment by Sayamame-beans** [comment]
とりあえずマージしてリリースします？(忘れてた)

**Comment by anatawa12** [comment]
してよさそう

**Pull request merged**